contrib/gin-gonic/gin/gintrace.go

```diff
@@ -53,6 +54,11 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 		// pass the span through the request context
 		c.Request = c.Request.WithContext(ctx)
 
+		// Use AppSec if enabled by user
+		if appsec.Enabled() {
+			useAppSec(c, span)
+		}
+
 		// serve the request to the next middleware
 		c.Next()
```

Comment:
I don't think the withAppSec logic can be applied here, since we don't have a proper way to wrap a call to next, unlike for echo.
I'd argue that to avoid the repeated call we could perform the call in gintrace.go only if appsec is disabled.